server/src/main/java/io/druid/curator/discovery/DiscoveryModule.java

```diff
@@ -476,4 +507,34 @@ public void close() throws IOException
       // nothing
     }
   }
+
+  private static class DruidLeaderSelectorProvider implements Provider<DruidLeaderSelector>
```

Comment:
this is a utility class only and constructor is explicitly call to provide latchPath which are not part of guice binding , other things are injected and that is why it looks a bit different.